overlord/snapstate/storehelpers.go

```diff
@@ -108,11 +103,17 @@ func installInfo(st *state.State, name, channel, cohort string, revision snap.Re
 	action := &store.SnapAction{
 		Action:       "install",
 		InstanceName: name,
+	}
+
+	// cannot specify both with the API
+	if revOpts.Revision.Unset() {
 		// the desired channel
-		Channel: channel,
+		action.Channel = revOpts.Channel
+		// the desired cohort key
+		action.CohortKey = revOpts.CohortKey
+	} else {
 		// the desired revision
-		Revision:  revision,
-		CohortKey: cohort,
+		action.Revision = revOpts.Revision
```

Comment:
Wondered whether we should drop some internal error if cohort key is set, but there's a similar check before we reach this place already, so probably not worth it.